app/uk/gov/hmrc/agentsubscriptionfrontend/controllers/StartController.scala

```diff
@@ -39,7 +39,8 @@ class StartController @Inject()(
   val metrics: Metrics,
   override val appConfig: AppConfig,
   sessionStoreService: SessionStoreService,
-  subscriptionService: SubscriptionService)(implicit val aConfig: AppConfig)
+  subscriptionService: SubscriptionService,
+  subscriptionCtrlr: SubscriptionController)(implicit val aConfig: AppConfig)
```

Comment:
you're importing a big controller here, rather not do that if we can. Something to discuss about in the workshop about design patterns. use reverse routing for this.

Reply:
Yes, I wasn't happy with this either, controllers pulling in other controllers as dependencies. 
The only thing it uses is the `SubscriptionController.redirectUponSuccessfulSubscription` funnction which is re-used in 3 different places in two different controllers and at least keeps in one place all the routing for whether the `/link-account` page should be shown or not.
It should live somewhere else.

Reply:
reverse routing is fine, you're only redirecting, so there is no need to import this SubscriptionController

Reply:
Captured in tech debt ticket ASA-69